satpy/readers/gms/gms5_vissr_l1b.py

```diff
@@ -0,0 +1,803 @@
+
+.. code-block:: none
+
+    VISSR_20020101_0031_IR1.A.IMG
+    VISSR_20020101_0031_IR2.A.IMG
+    VISSR_20020101_0031_IR3.A.IMG
+    VISSR_20020101_0031_VIS.A.IMG
+
+This is how to read them with Satpy:
+
+.. code-block:: python
+
+    from satpy import Scene
+    import glob
+
+    filenames = glob.glob(""/data/VISSR*")
+    scene = Scene(filenames, reader="gms5-vissr_l1b")
+    scene.load(["VIS", "IR1"])
+
+
+References
+~~~~~~~~~~
+
+Details about platform, instrument and data format can be found in the
+following references:
+
+    - `VISSR Format Description`_
+    - `GMS User Guide`_
+
+.. _VISSR Format Description:
+    https://www.data.jma.go.jp/mscweb/en/operation/fig/VISSR_FORMAT_GMS-5.pdf
+.. _GMS User Guide:
+    https://www.data.jma.go.jp/mscweb/en/operation/fig/GMS_Users_Guide_3rd_Edition_Rev1.pdf
+
+
+Compression
+-----------
+
+Gzip-compressed VISSR files can be decompressed on the fly using
+:class:`~satpy.readers.FSFile`:
+
+.. code-block:: python
+
+    import fsspec
+    from satpy import Scene
+    from satpy.readers import FSFile
+
+    filename = "VISSR_19960217_2331_IR1.A.IMG.gz"
+    open_file = fsspec.open(filename, compression="gzip")
+    fs_file = FSFile(open_file)
+    scene = Scene([fs_file], reader="gms5-vissr_l1b")
+    scene.load(["IR1"])
+
+
+Calibration
+-----------
+
+Sensor counts are calibrated by looking up reflectance/temperature values in the
+calibration tables included in each file. See section 2.2 in the VISSR user
+guide.
+
+
+Navigation
+----------
+
+VISSR images are oversampled and not rectified.
+
+
+Oversampling
+~~~~~~~~~~~~
+VISSR oversamples the viewed scene in E-W direction by a factor of ~1.46:
+IR/VIS pixels are 14/3.5 urad on a side, but the instrument samples every
+9.57/2.39 urad in E-W direction. That means pixels are actually overlapping on
+the ground.
+
+This cannot be represented by a pyresample area definition, so each dataset
+is accompanied by 2-dimensional longitude and latitude coordinates. For
+resampling purpose a full disc area definition with uniform sampling is provided
+via
+
+.. code-block:: python
+
+    scene[dataset].attrs["area_def_uniform_sampling"]
+
+
+Rectification
+~~~~~~~~~~~~~
+
+VISSR images are not rectified. That means lon/lat coordinates are different
+
+1) for all channels of the same repeat cycle, even if their spatial resolution
+   is identical (IR channels)
+2) for different repeat cycles, even if the channel is identical
+
+However, the above area definition is using the nominal subsatellite point as
+projection center. As this rarely changes, the area definition is pretty
+constant.
+
+
+Performance
+~~~~~~~~~~~
+
+Navigation of VISSR images is computationally expensive, because for each pixel
+the view vector of the (rotating) instrument needs to be intersected with the
+earth, including interpolation of attitude and orbit prediction. For IR channels
+this takes about 10 seconds, for VIS channels about 160 seconds.
```

Comment:
Is this still accurate after you optimization?

Reply:
Yes, unfortunately :)

Reply:
But this is comparable with JMA's C library now.